src/middlewared/middlewared/plugins/pool_/dataset.py

```diff
@@ -657,6 +662,24 @@ async def do_create(self, app, data):
             )
             await acl_job.wait()
 
+        elif data['type'] == 'FILESYSTEM' and data['share_type'] == 'APPS':
+            acl = (await self.middleware.call('filesystem.acltemplate.by_path', {
+                'query-filters': [('name', '=', 'NFS4_RESTRICTED')],
+                'format-options': {'canonicalize': True, 'ensure_builtins': True},
+            }))[0]['acl']
+            acl.append({
+                'tag': 'USER',
+                'id': 568,
+                'perms': {'BASIC': 'MODIFY'},
+                'flags': {'BASIC': 'INHERIT'},
+                'type': 'ALLOW'
+            })
+            acl_job = await self.middleware.call('filesystem.setacl', {
+                'path': mountpoint,
+                'dacl': acl,
+            })
+            await acl_job.wait(raise_error=True)
```

Comment:
Should we raise error at L663 too ?

Reply:
I refactored slightly so that we only have one acl_job now. We also now perform execute check on mountpoint and raise ValidationError if the ACL contains entries that won't be effective.